File: app/src/main/java/com/example/cookbook/recipesPage/RecipesListAdapter.java
package com.example.cookbook.recipesPage;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.cookbook.R;

import java.util.Random;

public class RecipesListAdapter extends RecyclerView.Adapter<RecipesListViewHolder> {

    private Context mContext;
    private Random mRandom;

    @NonNull
    @Override
    public RecipesListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        mContext = parent.getContext();
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        View view = layoutInflater.inflate(R.layout.recyclerview_all_recipes_item, parent, false);
        return new RecipesListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecipesListViewHolder holder, int position) {
        mRandom = new Random();
        holder.mPhotoContainer.getLayoutParams().height = getRandomHeight(550, 800);
    }

    @Override
    public int getItemCount() {
        return 10;
    }

    public int getRandomHeight(int min, int max) {
        return mRandom.nextInt((max - min) + 1) + min;
    }
}

File: app/src/main/java/com/example/cookbook/recipesPage/RecipesListAdapter.kt
package com.example.cookbook.recipesPage

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.recyclerview.widget.RecyclerView
import com.example.cookbook.R
import com.example.cookbook.models.Recipe
import java.util.*

class RecipesListAdapter constructor(list:MutableList<Recipe>) : RecyclerView.Adapter<RecipesListViewHolder>() {
    private var mContext: Context? = null
    private var mRandom: Random? = null
    private var mRecipeList:MutableList<Recipe> = list


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecipesListViewHolder {
        mContext = parent.context
        val layoutInflater = LayoutInflater.from(mContext)
        val view = layoutInflater.inflate(R.layout.recyclerview_all_recipes_item, parent, false)
        return RecipesListViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecipesListViewHolder, position: Int) {
        mRandom = Random()
        holder.mPhotoContainer.layoutParams.height = getRandomHeight(550, 800)
        holder.updateRecipeCardUi(mRecipeList[position])
    }

    override fun getItemCount(): Int {
        return mRecipeList.size
    }

    private fun getRandomHeight(min: Int, max: Int): Int {
        return mRandom!!.nextInt(max - min + 1) + min
    }

    fun updateList(newList:List<Recipe>){
        mRecipeList.clear()
        mRecipeList.addAll(newList)
        notifyDataSetChanged()
    }
}